lib/modules/datasource/generic-json/readme.md

```diff
@@ -0,0 +1,133 @@
+The `generic-json` datasource allows to gather dependency information from generic endpoints or files.
+These files or endpoints have to contain or return valid json objects in the following format:
+
+Minimal supported returned object:
+
+```json
+{
+  "releases": [
+    {
+      "version": "v1.1.0"
+    },
+    {
+      "version": "v1.2.0"
+    }
+  ]
+}
+```
+
+Complete supported object
+
+```json
+{
+  "releases": [
+    {
+      "version": "v1.0.0",
+      "isDeprecated": true,
+      "releaseTimestamp": "2022-12-24T18:21Z",
+      "changelogUrl": "https://github.com/demo-org/demo/blob/main/CHANGELOG.md#v0710",
+      "sourceUrl": "https://github.com/demo-org/demo",
+      "sourceDirectory": "monorepo/folder"
+    }
+  ],
+  "sourceUrl": "https://github.com/demo-org/demo",
+  "sourceDirectory": "monorepo/folder",
+  "changelogUrl": "https://github.com/demo-org/demo/blob/main/CHANGELOG.md",
+  "homepage": "https://demo.org"
+}
+```
+
+`packageName` will be interpreted as [JSONata](https://jsonata.org/) query.
+`registryUrl` is used to define an endpoint ( prefixed with `http://` or `https://`) or a local file ( prefixed with `file://`)
```

Comment:
I also don't understand this

Reply:
So we are supporting local file-based datasources? That needs a security check if so

Reply:
local files should be hidden behind a feature flag. otherwise we probably allow a repo to read arbitrary files from bot

Reply:
ok. it's useing `readLocalFile` so it can only read files from current repo. so no security implications yet